### PR TITLE
Support sasl-plain username password in text file

### LINKED ISSUE #186: do not store sasl password in process state
the password might get dumped to crash logs.

### PR DESCRIPTION
**Pull request by zmstone**
fixes #186

### CODE REVIEW

File: src/brod_utils.erl [hunk 1]
```diff
@@ -210,9 +212,55 @@ fetch(SockPid, ReqFun, MaxBytes, Offset) ->
...
+%% the plain text being dumped to crash logs
+%% @end
+-spec init_sasl_opt(client_config()) -> client_config().
+init_sasl_opt(Config) ->
+  case proplists:get_value(sasl, Config) of
+    {plain, User, Pass} when is_list(Pass) orelse is_binary(Pass) ->
+      replace_prop(sasl, {plain, User, fun() -> Pass end}, Config);
+    {plain, File} ->
+      {User, Pass} = read_sasl_file(File),
+      replace_prop(sasl, {plain, User, fun() -> Pass end}, Config);
```
Comment (id) [inline src/brod_utils.erl]: lists:keystore?

File: src/brod_utils.erl [hunk 2]
```diff
@@ -210,9 +212,55 @@ fetch(SockPid, ReqFun, MaxBytes, Offset) ->
       end
   end.
 
+%% @doc Get sasl options from client config.
+-spec get_sasl_opt(client_config()) -> sasl_opt().
+get_sasl_opt(Config) ->
+  case proplists:get_value(sasl, Config) of
+    {plain, User, PassFun} when is_function(PassFun) ->
+      {plain, User, PassFun()};
+    {plain, File} ->
```
Comment (id) [inline src/brod_utils.erl:221]: in which situation this case will trigger?

Comment (zmstone) [reply src/brod_utils.erl:221]: brod-cli. or when calling `brod:connect_leader/4` directly or indirectly e.g. from `brod:fetch/8`

Comment (id) [reply src/brod_utils.erl:221]: get_sasl_opt and init_sasl_opt are mostly identical. Perhaps leave just this one?

Comment (zmstone) [reply src/brod_utils.erl:221]: init is to hide password in anonymous fun
get is to evaluate the anonymous fun

File: src/brod_utils.erl (hunk 1 above)
Comment (zmstone) [reply src/brod_utils.erl]: we use `proplists:` APIs everywhere for client config

File: src/brod_utils.erl (hunk 2 above)
Comment (id) [reply src/brod_utils.erl:221]: if you call `brod:connect_leader/4 ` directly then you don't have a fun to evaluate, or?

Comment (id) [reply src/brod_utils.erl:221]: nvm, then it will pass User/Pass as is.

**Review by id — APPROVED** [review]

**Pull request merged**